sycl/test/basic_tests/marray/marray.cpp

```diff
@@ -0,0 +1,87 @@
+// RUN: %clangxx %s -o %t.out -lsycl -I %sycl_include
+// RUN: %t.out
+//==--------------- marray.cpp - SYCL marray test --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+using namespace sycl;
+
+int main() {
+  // Constructing vector from a scalar
+  sycl::marray<int, 1> marray_from_one_elem(1);
+
+  // Check broadcasting operator=
+  sycl::marray<float, 4> b_marray(1.0);
```

Comment:
Probably in the following we need more `const` or rvalue to test that it can compile in that case...